applications/nrf_desktop/src/hw_interface/motion_buttons.c

```diff
@@ -33,11 +41,26 @@ enum dir {
 	DIR_COUNT
 };
 
+BUILD_ASSERT(DIR_COUNT <= 8);
+static uint8_t active_dir_bm;
+
+static motion_ts timestamp[DIR_COUNT];
+static int32_t x_remainder;
+static int32_t y_remainder;
```

Comment:
I wanted to allow for negative reminders, so that I used signed types. I would say they should be acceptable, because I think `sys_clock_hw_cycles_per_sec` would rather not be that big.